app/services/pagarme/api.rb

```diff
@@ -45,8 +45,60 @@ def create_credit_card(attributes)
       raise InvalidAttributeError, e.message
     end
 
+    def retrieve_pagarme_subscriptions(subscriptions_ids)
+      subscription_request = subscriptions_params(subscriptions_ids)
+
+      response = pagarme_request(subscription_request)
+
+      Pagarme::ResponseParser.process(response.body)
+    end
+
+    def cancel_subscriptions(canceled_subscriptions)
+      cancel_request = cancel_params(canceled_subscriptions)
+      pagarme_request(cancel_request)
+    end
+
     private
 
+    def pagarme_request(request)
```

Comment:
For me is not so clear what request is this, just reading the method you can't tell if it's a post or get, etc and which endpoint you are requesting.